lib/zclient.h

```diff
@@ -458,7 +469,7 @@ struct zapi_route {
 #define ZEBRA_FLAG_RR_USE_DISTANCE    0x40
 
 	/* The older XXX_MESSAGE flags live here */
-	uint8_t message;
+	uint32_t message;
```

Comment:
so it looks as if this is only used to indicate that nexthops have an extra attribute, but using this route-level flag (and changing its size) causes quite a bit of change to ripple out. why not use a nexthop-level flag instead - that would reduce the impact?

Reply:
@rwestphal IIRC we were actually aware of this problem and simply forgot about this. Thanks for pointing this out @mjstapp

Reply:
The problem is that we use `zapi_route` (and the associated encode/decode functions) to send/receive nexthop update messages too (it's a multi-purpose message format). And for nexthop updates we need a single color, which combined with the destination prefix form an SR-TE endpoint. For routes, however, we need per-nexthop colors (since different nexthops can have different colors or have no color at all). So I think we need to change `zapi_route` to support both per-route and per-nexthop colors, as confusing as it can be.